metals/src/main/scala/scala/meta/internal/metals/codeactions/OrganizeImports.scala

```diff
@@ -85,33 +96,18 @@ class SourceOrganizeImports(
   override val kind: String = SourceOrganizeImports.kind
   override protected val title: String = SourceOrganizeImports.title
 
-  override protected def isCallAllowed(
+  override protected def disabledReason(
       file: AbsolutePath,
       params: CodeActionParams,
-  ): Boolean = {
-    val validCall = isScalaOrSbt(file) && isSourceOrganizeImportCalled(params)
-    if (validCall) {
-      if (diagnostics.hasDiagnosticError(file)) {
-        languageClient.showMessage(
-          l.MessageType.Warning,
-          s"Fix ${file.toNIO.getFileName} before trying to organize your imports",
-        )
-        scribe.info("Can not organize imports if file has error")
-        false
-      } else {
-        true
-      }
+  ): Option[String] =
+    if (!isScalaOrSbt(file)) {
+      Some("Only supported for Scala and sbt files")
+    } else if (diagnostics.hasDiagnosticError(file)) {
+      scribe.info(canNotOrganizeImportsWithErrors)
+      Some(canNotOrganizeImportsWithErrors)
     } else {
```

Comment:
We are now missing one of the conditions `isSourceOrganizeImportCalled(params)` and organize imports action is returned everywhere. We want to return it on unused imports and also when explicitly called.  Maybe we should still keep isCallAllowed for that and also check that flag?